src/extension/loadLogs.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/// <reference path="jsonSourceMap.d.ts" />
+import { execFileSync } from 'child_process';
+import * as fs from 'fs';
+import jsonMap from 'json-source-map';
+import { join } from 'path';
+import { Log } from 'sarif';
+import { eq, gt, lt } from 'semver';
+import { tmpNameSync } from 'tmp';
+import { ProgressLocation, Uri, window } from 'vscode';
+import { augmentLog, JsonMap } from '../shared';
+import { Store } from './store';
+
+export async function loadLogs(uris: Uri[], token?: { isCancellationRequested: boolean }) {
+    const logs = uris
+        .map(uri => {
+            if (token?.isCancellationRequested) return undefined;
+            try {
+                const file = fs.readFileSync(uri.fsPath, 'utf8')  // Assume scheme file.
+                    .replace(/^\uFEFF/, ''); // Trim BOM.
+                const {data: log, pointers} = jsonMap.parse(file) as { data: Log, pointers: JsonMap};
+                log._uri = uri.toString();
+                log._jsonMap = pointers;
+                return log;
+            } catch (error) {
+                window.showErrorMessage(`Failed to parse '${uri.fsPath}'`);
+                return undefined;
+            }
+        })
+        .filter(log => log) as Log[];
```

Comment:
>filter(log => log) [](start = 9, length = 18)

necessary?

Reply:
We want to filter out the undefined array values. These value were produced on line 29. This expression is idiomatic for removing "empty" values and is used consistently across this codebase.

---
In reply to: [441209455](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r441209455) [](ancestors = 441209455)